### DISCUSSION TITLE
Rendering is not updating when dynamically changing data in ElTreeV2

### DISCUSSION THREAD

**Discussion by sb-s-makino** [element-plus/element-plus#13686] [Q&A]
I added the ElTreeV2 component inside a custom component called CustomTree. Then, I placed the CustomTree component inside the TreeView component and passed the tree structure data through props. Subsequently, I dynamically changed the data. As shown in the bottom left of the illustration, items with labels "abc" and "xyz" were added, and the number of top-level elements increased from 3 to 5. As visible in Vue DevTools (on the right side of the attachment), the data of ElTreeV2 has been updated. However, the data of ElFixedSizeList below it is not being updated.

How can I update the rendering in this scenario?

```
<template>
  <el-tree-v2
    :data="props.data"
    :height="400"
    class="mt-4 border-2 border-slate-300"
    @node-click="clickHandler"
  >
    <template #default="{ node }">
      <font-awesome-icon v-if="!node.isLeaf" icon="fa-folder" class="pr-2" />
      <span>{{ node.label }}</span>
    </template>
  </el-tree-v2>
  <div>{{props.data}}</div>
</template>

<script setup lang="ts" >
import { reactive, watch } from 'vue'
import { ElTreeV2 } from 'element-plus'
const props = defineProps({
  data: {
    type: Array,
    required: true
  }
})

const emit = defineEmits(['selected'])

const clickHandler = (data: TreeNodeData, node: TreeNode) => {
  const isFolder = !node.isLeaf
  emit('selected', { id: data.id, label: data.label, isFolder })
}
</script>

<style></style>
```
<img width="1818" alt="スクリーンショット 2023-07-21 19 06 11" src="https://github.com/element-plus/element-plus/assets/121912937/ef7b387c-c755-4e64-9864-efd7ef5e8a4a">

**Comment by pitang1965** [comment]
I tried adding the following code to the code displayed in [Edit in Playground] below, but it did not work.

https://element-plus.org/en-US/component/tree.html#basic-usage

Here is the code to add

```
:

<button @click="add">Add Level one 4</button>

:

const add =() => {
  data.push({
      label: "Level on 4",
      children: []
    })
  alert (JSON.stringify(data))
}

:
```

**Comment by moon-in-river** [comment]
![image](https://github.com/element-plus/element-plus/assets/55790540/1b1f0839-5b0f-4e28-8f81-f44507cf29c3)
ElTreeV2 data changed，but ElFixedSizeList data[index].data not change, maybe reason? how to resolve?
